storage/replica.go

```diff
@@ -3177,6 +3177,12 @@ func (r *Replica) executeBatch(
 		optimizePuts(batch, ba.Requests, ba.Header.DistinctSpans)
 	}
 
+	// Update the node clock with the serviced request. This maintains a high
+	// water mark for all ops serviced, so that received ops without a timestamp
+	// specified are guaranteed one higher than any op already executed for
+	// overlapping keys.
+	r.store.Clock().Update(ba.Header.Timestamp)
```

Comment:
Ah, got it. That check seems to fail independent of this change. Perhaps you'd like to debug what is going on there?